erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```
@@ -2097,6 +2099,112 @@ def set_status(self, update=False, status=None, update_modified=True):
+		"""
+		if not self.items:
+			return
+
+		# Get all unique Sales Orders affected by this invoice
+		affected_sales_orders = set()
+		items_with_discounts = []
+
+		for item in self.items:
+			if not item.so_detail:
+				continue
+			
```

Ruff (0.12.2)

Blank line contains whitespace

Remove whitespace from blank line

(W293)